Firmware/mmu2.cpp

```diff
@@ -679,24 +679,23 @@ void MMU2::CheckUserInput() {
         lastButton = Buttons::NoButton; // Clear it.
     }
 
+    if (mmu2.MMULastErrorSource() == MMU2::ErrorSourcePrinter && btn != Buttons::NoButton)
+    {
+        // When the printer has raised an error screen, and a button was selected
+        // the error screen should always be dismissed.
+        ClearPrinterError();
```

Comment:
**Note:**
Right now I need this change, because `lastErrorCode` must be cleared. We cannot rely on checking the current error code from the MMU protocol since Printer errors do not originate from the protocol.

Consider the following scenario:

1.  `lastErrorCode` starts with value `ErrorCode::OK`
2. The firmware raises `FILAMENT_CHANGE` error. It's a new error code so `lastErrorCode = ErrorCode::FILAMENT_CHANGE`
3. The user clicks Load and the error screen is dismissed.
4. FINDA runout triggers M600 and the firmware raises `FILAMENT_CHANGE` error.
5. `lastErrorCode` is still `ErrorCode::FILAMENT_CHANGE`. This results in the error not being logged fully. Also implies `IncrementMMUFails()` would not be called if one got the same error twice e.g. failed load!

The purpose of `lastErrorCode` in my opinion is to prevent the error reporting to flood the serial log _while_ the error is unresolved. When the MMU reports an error, it will continue to keep reporting the same error every second. This we don't want in the logs. BUT, if the same error is resolved twice (e.g. failed load) I would expect to see two logged error events since they are independent. Also I would expect the statistics in the menus to reflect this as well, that is => counting two failures and not one.

Proposal: Set `lastErrorCode` to `ErrorCode::OK` when the error is resolved.